files/ansible/library/test_result_parser.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: test_result_parser
+
+short_description: This module parses test output with regexes
+and puts the results into a Junit/Xunit XML files.
+
+version_added: "2.4"
+
+description:
+    - "This is my longer description explaining my sample module"
+
+options:
+    test_output_path:
+        description:
+            - This is the path to the file containing the test's output.
+        required: true
+    test_threshold:
+        description:
+            - This is the value that will determine the test result (via the
+            failed method).
+        required: true
+    json_or_xml:
+        description:
+            - This determines whether the parsed result is outputted in XML or
+            JSON. Currently JSON is not available, XML is default.
+        required: false
+    junit_write_path:
+        description:
+            - This determines where the parsed JUnit/Xunit file will be
+            written.
+        required: false
+    extra_flags:
+        description:
+            - This determines the tag on the filename.
```

Comment:
extra_flags is a bad name if it only has one reason. Maybe start with "tag" and expand later?